### PR TITLE
Add OTel Agent e2e pipeline tests

### PR DESCRIPTION
**Pull request by liustanley**
### What does this PR do?
This PR adds e2e pipeline tests for the OTel Agent and refactors the file structure of `new-e2e/tests/otel` according to our [testing plan](https://docs.google.com/document/d/13ddBAG48Wxzbx2snuzv-xKlC2MJSn87ORw53ivqLAhI/edit#heading=h.24o4ezp4rjwn).

These tests use four different collector config files:
- `complete.yml` for a complete config file to use with the converter disabled
- `minimal.yml` for a minimal config file to test that the converter adds components as expected
- `sampling.yml` for adding the probabilistic sampler and verifying APM stats still work
- `ia.yml` for a minimal config file to test the infra attributes processor with high cardinality

The following test cases are implemented in `utils/pipelines_utils.go`:
- Verify fakeintake receives scraped prometheus metrics: collector metrics, trace agent metrics, and agent metrics
- Verify fakeintake receives telemetry from telemetrygen
  - Resource attributes are tagged as expected
  - Telemetry attributes are tagged as expected
  - Infraattributesprocessor adds kubernetes tags as expected
  - Service, env, and version tags appear as expected
  - Host appears as expected and consistent across all signals
  - Trace metrics are received by fakeintake
- Verify trace metrics work with sampling as expected
   

### Additional Notes

The non-pipeline tests have been moved to `utils/config_utils.go`, and will be expanded on in a future PR for adding extensive config tests.

### CODE REVIEW

File: test/new-e2e/tests/otel/otel-agent/complete_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package otelagent contains e2e otel agent tests
+package otelagent
+
+import (
+	_ "embed"
```
Comment (jackgopack4) [inline test/new-e2e/tests/otel/otel-agent/complete_test.go:10]: asking out of ignorance, why is this added as "_"?

File: test/new-e2e/tests/otel/utils/config_utils.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+)
+
+// TestOTelAgentInstalled checks that the OTel Agent is installed in the test suite
+func TestOTelAgentInstalled(s OTelTestSuite) {
+	agent := getAgentPod(s)
+	assert.Contains(s.T(), agent.ObjectMeta.String(), "otel-agent")
+}
```
Comment (dineshg13) [inline test/new-e2e/tests/otel/utils/config_utils.go:24]: should we verify version ?

File: test/new-e2e/tests/otel/utils/pipelines_utils.go [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+		assert.Equal(s.T(), "telemetrygen", sp.Meta["otel.library.name"])
+		assert.Equal(s.T(), sp.Meta["k8s.node.name"], tp.Hostname)
+
+		// Verify container tags from infraattributes processor
+		assert.NotNil(s.T(), ctags["kube_container_name"])
+		assert.NotNil(s.T(), ctags["kube_namespace"])
+		assert.NotNil(s.T(), ctags["pod_name"])
+		assert.Equal(s.T(), sp.Meta["k8s.container.name"], ctags["kube_container_name"])
+		assert.Equal(s.T(), sp.Meta["k8s.namespace.name"], ctags["kube_namespace"])
+		assert.Equal(s.T(), sp.Meta["k8s.pod.name"], ctags["pod_name"])
```
Comment (dineshg13) [inline test/new-e2e/tests/otel/utils/pipelines_utils.go:88]: We set these on SDK, can we test specific ones that infra attributes processor adds ?

**Review by dineshg13 — COMMENTED** [review]
We also need to test infra attributes processor with different cardinalities

File: test/new-e2e/tests/otel/otel-agent/config/sampling.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      compute_top_level_by_span_kind: true
+      peer_tags_aggregation: true
+service:
+  pipelines:
+    traces:
+      receivers: [otlp]
+      processors: [batch]
+      exporters: [datadog/connector]
+    traces/send:
+      receivers: [datadog/connector]
```
[the remark this answers is not in the document]
Comment (liustanley) [reply test/new-e2e/tests/otel/otel-agent/config/sampling.yml:36]: There's another trace pipeline with the OTLP receiver, does this one need it too?

File: test/new-e2e/tests/otel/otel-agent/complete_test.go (hunk 1 above)
Comment (liustanley) [reply test/new-e2e/tests/otel/otel-agent/complete_test.go:10]: We're only using this package to embed yaml files into strings (`//go:embed config/complete.yml`), but aren't using any of its functions so it needs to be a blank import: https://pkg.go.dev/embed#hdr-Strings_and_Bytes.

File: test/new-e2e/tests/otel/utils/pipelines_utils.go (hunk 3 above)
Comment (liustanley) [reply test/new-e2e/tests/otel/utils/pipelines_utils.go:88]: These are all the tags added by the infra attributes processor, if we want the infra attributes processor to add more tags we need to set more Kubernetes attributes on the SDK
```
tags:"custom.attribute:true" tags:"deployment.environment:e2e" tags:"host.name:kind-control-plane" tags:"k8s.container.name:telemetrygen-job" tags:"k8s.namespace.name:datadog" tags:"k8s.node.name:kind-control-plane" tags:"k8s.pod.name:telemetrygen-job-metrics-fv6tn" tags:"service.name:telemetrygen-job" tags:"service.version:1.0" tags:"env:e2e" tags:"service:telemetrygen-job" tags:"version:1.0" tags:"kube_container_name:telemetrygen-job" tags:"kube_namespace:datadog" tags:"pod_name:telemetrygen-job-metrics-fv6tn"
```

File: test/new-e2e/tests/otel/utils/config_utils.go (hunk 2 above)
Comment (liustanley) [reply test/new-e2e/tests/otel/utils/config_utils.go:24]: I tried looking in this agent object and didn't see any fields for version. There are some fields for image but that depends on if we're running the tests locally (gcr.io/datadoghq/agent:7-ot-beta) or if it's running in a PR (uses PR image).

**Review by KevinFairise2 — COMMENTED** [review]
Some suggestions to avoid panic when accessing empty array. But otherwise lgtm!

File: test/new-e2e/tests/otel/utils/pipelines_utils.go [hunk 5]
```diff
@@ -0,0 +1,580 @@
...
+	}
+	splits := strings.Split(ctags, ",")
+	m := make(map[string]string)
+	for _, s := range splits {
+		kv := strings.SplitN(s, ":", 2)
+		if !assert.Len(t, kv, 2, "malformed container tag: %v", s) {
+			continue
+		}
+		m[kv[0]] = kv[1]
+	}
```
Comment (dineshg13) [inline test/new-e2e/tests/otel/utils/pipelines_utils.go:568]: can call getTagMapFromSlice

**Review by dineshg13 — APPROVED** [review]
LGTM . There few things we can handle in next iterations 
1. Use Yaml strings for defining deployments 
2. Can we extract code and create some common utils esp `complete_test.go` & `infraatributes_test.go` , etc have most common code

**Review by KevinFairise2 — APPROVED** [review]
